test/index.html

```diff
@@ -0,0 +1,247 @@
+                         next : function(){ return impress().next(); },
+                         text : "next() wraps around to first step (4->1)" },
+                       { me   : step1,
+                         next : function(){ return impress().prev(); },
+                         text : "prev() wraps around to last step (1->4)" },
+                       { me   : step4,
+                         next : function(){ return impress().prev(); },
+                         text : "prev() called and returns ok (4->3)" },
+                       { me   : step3,
+                         next : false }      
+      ];
+      
+      var assertStepEnter = function( event, registeredListener ) {
+        assert.equal( event.target, nextStep[i].me,
+                      event.target.id + " triggered impress:stepenter event." );
+        assert.ok( event.target.classList.contains("active"),
+                   event.target.id + " set active css class." );
+        assert.ok( event.target.classList.contains("present"),
+                   event.target.id + " set present css class." );
+        assert.ok( !event.target.classList.contains("future"),
+                   event.target.id + " unset future css class." );
+        assert.ok( !event.target.classList.contains("past"),
+                   event.target.id + " unset past css class." );
+
+        if( nextStep[i].next ) {
+          assert.ok( nextStep[i].next(), nextStep[i].text );
+          i++;
+        } else {
+          // Remember to cleanup, since we're operating outside of qunit-fixture
+          root.removeEventListener( "impress:stepenter", assertStepEnterWrapper );
+          doneWithNext();
+        }
+      };
+      
+      var assertStepEnterWrapper = function( event ) {
+        setTimeout( function() { assertStepEnter( event ) },
+                    wait*2 ); // wait*2 ensures that assertStepLeave is finished 
+                              // first, since here we'll call nextStep.next and i++
+      };
+      root.addEventListener( "impress:stepenter", assertStepEnterWrapper );
+
+
+      var prevStep = [ step1 ];
+      for(j=0; j < nextStep.length; j++){
+        prevStep.push( nextStep[j].me );
+      }
+      var assertStepLeave = function( event, registeredListener ) {
+        assert.equal( event.target, prevStep[i],
+                      event.target.id + " triggered impress:stepleave event." );
+        assert.ok( !event.target.classList.contains("active"),
+                   event.target.id + " unset active css class." );
+        assert.ok( !event.target.classList.contains("present"),
+                   event.target.id + " unset present css class." );
+        assert.ok( !event.target.classList.contains("future"),
+                   event.target.id + " unset future css class." );
+        assert.ok( event.target.classList.contains("past"),
+                   event.target.id + " set past css class." );
+        
+        if( i+2 == prevStep.length ) {
```

Comment:
Btw, if you wanted to focus on testing each event listener separately, it still leaves an open question: where would you check for the "active" css class? It is set inside the goto(), so not set inside either event listener. So should there be yet another test where that is checked separately?

This is just to underline my point: This test is not checking event triggers per se, rather it is testing everything related to moving from one step to next.